### PR TITLE
dev display limit amount while 'abonnement'

### PR DESCRIPTION
**Pull request by Raulnet**
dev display limit amount while 'abonnement'

### CODE REVIEW

File: src/Validator/MaxMonthDonationValidator.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+use AppBundle\Donation\DonationRequest;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+class MaxMonthDonationValidator extends ConstraintValidator
+{
+    /**
+     * @param DonationRequest  $value
+     * @param MaxMonthDonation $constraint
```
[the remark this answers is not in the document]
Comment (Raulnet) [reply src/Validator/MaxMonthDonationValidator.php]: added

File: translations/validators.fr.yml [hunk 2]
```diff
@@ -122,6 +122,7 @@ turnkey_project_file.valid_mimetypes: Seuls les fichiers PDF, Word et Powerpoint
 donation.subscription.not_unique: Vous faites déjà un don mensuel à La République En Marche ! Vous pouvez <a href="{{ profile_url }}">vous rendre sur votre profil</a> pour l’annuler ou <a href="{{ donation_url }}">faire un nouveau don unique</a>.
 donation.subscription.not_unique_from_anonymous: Vous faites déjà un don mensuel à La République En Marche ! Vous pouvez <a href="https://contact.en-marche.fr/">nous contacter pour l’annuler</a> ou <a href="{{ donation_url }}">faire un nouveau don unique</a>.
 donation.max_fiscal_year_donation: Le don que vous vous apprêtez à faire est trop élevé, car vous avez déjà donné {{ total_current_amount }} euros cette année. Les dons étant limités à {{ max_amount_per_fiscal_year }} euros par an et par personne, vous pouvez encore donner {{ max_donation_remaining_possible }} euros.
+donation.max_fiscal_month_donation: Votre don mensuel ne peut dépasser 625 euros. Si vous souhaitez donner plus, vous pourrez compléter avec un don ponctuel.
```
[the remark this answers is not in the document]
Comment (Raulnet) [reply translations/validators.fr.yml]: edited

File: src/Validator/MaxMonthDonationValidator.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        }
+
+        if (!$value instanceof DonationRequest) {
+            throw new UnexpectedTypeException($value, DonationRequest::class);
+        }
+
+        if ($value->isSubscription() && $value->getAmount() * 100 > $constraint->maxDonationInCents) {
+            $this->context
+                ->buildViolation($constraint->message)
+                ->addViolation()
```
[the remark this answers is not in the document]
Comment (Raulnet) [reply src/Validator/MaxMonthDonationValidator.php]: fixed

File: front/components/DonationAmountChooser.js [hunk 4]
```diff
@@ -22,17 +25,53 @@ export default class DonationAmountChooser extends React.Component {
...
                 <AmountChooser
                     name={this.props.name}
                     value={this.props.value}
-                    abonnement={this.props.abonnement}
                     onChange={this.handleAmountChange}
+                    maxValue={this.state.abonnement ? maxAbonnementAmount : maxAmount}
                 />
 
+                {this.state.abonnement && this.state.amount > maxAbonnementAmount &&
+                <div className="amount-chooser__help">
```
Comment (ottaviano) [inline front/components/DonationAmountChooser.js]: can you indent here pls?

Comment (Raulnet) [reply front/components/DonationAmountChooser.js]: indented

**Pull request merged**